### PR TITLE
[ticket/15201] Set user style to default style when it is disabled or uninstalled

### PR DESCRIPTION
**Pull request by rubencm**
PHPBB3-15201

Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [ ] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15201

### CODE REVIEW

File: phpBB/includes/acp/acp_styles.php [hunk 1]
```diff
@@ -385,7 +385,7 @@ protected function action_deactivate()
 
 		// Reset default style for users who use selected styles
 		$sql = 'UPDATE ' . USERS_TABLE . '
-			SET user_style = 0
+			SET user_style = ' . $this->default_style . '
```
Comment (Nicofuma) [inline phpBB/includes/acp/acp_styles.php]: should be cast to int

File: phpBB/includes/acp/acp_styles.php [hunk 2]
```diff
@@ -1231,7 +1231,7 @@ protected function uninstall_style($style)
 
 		// Change default style for users
 		$sql = 'UPDATE ' . USERS_TABLE . '
-			SET user_style = 0
+			SET user_style = ' . $this->default_style . '
```
Comment (Nicofuma) [inline phpBB/includes/acp/acp_styles.php]: should be cast to int

**Comment by rubencm** [comment]
I'm not sure if the last commit breaks anything, but i didn't like that check that only exists because other parts of the code put the user style incorrectly

**Comment by rxu** [comment]
I'm not sure the code from the user class should be deleted as it may brake things for users having set outdated (deleted) styles.

**Comment by rubencm** [comment]
This should fix that:
https://github.com/rubencm/phpbb/blob/23992a0d515e31dc81a9fc03ffecdbd09f7d0615/phpBB/phpbb/db/migration/data/v310/style_update_p1.php#L186-L188
https://github.com/rubencm/phpbb/blob/23992a0d515e31dc81a9fc03ffecdbd09f7d0615/phpBB/phpbb/db/migration/data/v310/style_update_p1.php#L168

Or should i create another migration to be sure that no users have invalid styles?

For now i am trying to fix the travis errors, i dont know how to fix them.

**Comment by rxu** [comment]
That is fixing an old migration which won't affect already updated boards at all. New migration could, yes.

**Comment by rubencm** [comment]
Im leaving this issue free since i am not able to pass the tests

**Pull request closed without merging**